src/Microsoft.ML.Data/Transforms/SelectColumnsTransform.cs

```diff
@@ -0,0 +1,632 @@
+                _host = host;
+            }
+
+            public bool CanShuffle => Source.CanShuffle;
+
+            public ISchema Schema => _mapper.Schema;
+
+            public IDataView Source { get; }
+
+            public ISchema InputSchema => Source.Schema;
+
+            public long? GetRowCount(bool lazy = true)
+            {
+                return Source.GetRowCount(lazy);
+            }
+
+            public IRowCursor GetRowCursor(Func<int, bool> needCol, IRandom rand = null)
+            {
+                _host.AssertValue(needCol, "needCol");
+                _host.AssertValueOrNull(rand);
+
+                // Build out the active state for the input
+                var inputPred = GetDependencies(needCol);
+                var inputRowCursor = Source.GetRowCursor(inputPred, rand);
+
+                // Build the active state for the output
+                var active = Utils.BuildArray(Schema.ColumnCount, needCol);
+                return new RowCursor(_host, _mapper, inputRowCursor, active);
+            }
+
+            public IRowCursor[] GetRowCursorSet(out IRowCursorConsolidator consolidator, Func<int, bool> needCol, int n, IRandom rand = null)
+            {
+                _host.CheckValue(needCol, nameof(needCol));
+                _host.CheckValueOrNull(rand);
+
+                // Build out the active state for the input
+                var inputPred = GetDependencies(needCol);
+                var inputs = Source.GetRowCursorSet(out consolidator, inputPred, n, rand);
+
+                // Build out the acitve state for the output
+                var active = Utils.BuildArray(Schema.ColumnCount, needCol);
+                _host.AssertNonEmpty(inputs);
+
+                // No need to split if this is given 1 input cursor.
+                var cursors = new IRowCursor[inputs.Length];
+                for (int i = 0; i < inputs.Length; i++)
+                {
+                    cursors[i] = new RowCursor(_host, _mapper, inputs[i], active);
+                }
+                return cursors;
+            }
+
+            public void Save(ModelSaveContext ctx)
+            {
+                _transform.Save(ctx);
+            }
+
+            public Func<int, bool> GetDependencies(Func<int, bool> activeOutput)
+            {
+                var active = new bool[_mapper.InputSchema.ColumnCount];
+                var columnCount = _mapper.Schema.ColumnCount;
+                for (int colIdx = 0; colIdx < columnCount; ++colIdx)
+                {
+                    if (activeOutput(colIdx))
+                    {
+                        active[_mapper[colIdx]] = true;
+                    }
+                }
+
+                return col => active[col];
+            }
+
+            public IRow GetRow(IRow input, Func<int, bool> active, out Action disposer)
+            {
+                disposer = null;
+                using (var ch = _host.Start("GetEntireRow"))
+                {
```

Comment:
any reason to create it? I don't see any usage of it.

Reply:
no - I have removed it.


---
In reply to: [225346084](https://github.com/dotnet/machinelearning/pull/1269#discussion_r225346084) [](ancestors = 225346084)